Sources/QuadTreeNode.swift

```diff
@@ -0,0 +1,84 @@
+import MapKit
+
+// MARK: - Tree node
+
+final class QuadTreeNode {
+
+  private let capacity: Int
+  private let rect: MKMapRect
+  private var children: [QuadTreeNode] = []
+  private var annotations: [MKAnnotation] = []
+
+  var isLeaf: Bool {
+    return children.isEmpty
+  }
+
+  init(rect: MKMapRect, capacity: Int) {
+    self.rect = rect
+    self.capacity = capacity
+  }
+
+  @discardableResult
+  func add(annotation: MKAnnotation) -> Bool {
+    if !rect.contains(annotation: annotation) {
+      return false
+    }
+
+    if isLeaf {
+      annotations.append(annotation)
+      // Subdivide if needed
+      if annotations.count == capacity {
+        children = createChildren()
+      }
+    } else {
+      for child in children {
+        if child.add(annotation: annotation) {
+          return true
+        }
+      }
+
+      return false
+    }
+
+    return true
+  }
+
+  func annotations(inRect rect: MKMapRect) -> [MKAnnotation] {
+    if !self.rect.intersects(rect: rect) {
+      return []
+    }
+
+    var result: [MKAnnotation] = []
+
+    for annotation in annotations {
```

Comment:
Could you filter `annotations` before iterating over it so that you can skip the `if` statement inside?